packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,713 @@
+    addAll(children);
+  }
+
+  /// The associated ViewportOffset object for the viewport.
+  ViewportOffset get offset => _offset;
+  ViewportOffset _offset;
+  set offset(ViewportOffset value) {
+    assert(value != null);
+    if (value == _offset)
+      return;
+    if (attached)
+      _offset.removeListener(_hasScrolled);
+    _offset = value;
+    if (attached)
+      _offset.addListener(_hasScrolled);
+    markNeedsLayout();
+  }
+
+  /// {@template flutter.widgets.wheelList.diameterRatio}
+  /// A ratio between the diameter of the cylinder and the viewport's size
+  /// in the main axis.
+  ///
+  /// A value of 1 means the cylinder has the same diameter as the viewport's
+  /// size.
+  ///
+  /// A value smaller than 1 means items at the edges of the cylinder are
+  /// entirely contained inside the viewport.
+  ///
+  /// A value larger than 1 means angles less than +/- [math.pi] / 2 from the
+  /// center of the cylinder are visible.
+  ///
+  /// The same number of children will be visible in the viewport regardless of
+  /// the [diameterRatio]. The number of children visible is based on the
+  /// viewport's length along the main axis divided by the children's
+  /// [itemExtent]. Then the children are evenly distributed along the visible
+  /// angles up to +/- [math.pi] / 2.
+  ///
+  /// Just as it's impossible to stretch a paper to cover the an entire
+  /// half of a cylinder's surface where the cylinder has the same diameter
+  /// as the paper's length, choosing a [diameterRatio] smaller than [math.pi]
+  /// will leave same gaps between the children.
+  /// {@endtemplate}
+  double get diameterRatio => _diameterRatio;
+  double _diameterRatio;
+  set diameterRatio(double value) {
+    assert(value != null);
+    assert(
+      value > 0,
+      "You can't set a diameterRatio of 0. It would imply a cylinder of 0 "
+      'diameter in which case nothing will be drawn');
+    if (value == _diameterRatio)
+      return;
+    _diameterRatio = value;
+    _hasScrolled();
+  }
+
+  /// Perspective of the cylindrical projection. See doc in
+  /// [MatrixUtils.createCylindricalProjectionTransform].
+  double get perspective => _perspective;
+  double _perspective;
+  set perspective(double value) {
+    assert(value != null);
+    assert(
+      value <= 0.01,
+      'A perspective value too high will always be clipped in the z axis and unrenderable.\n'
+      'Choose a value lower than 0.01',
+    );
+    if (value == _perspective)
+      return;
+    _perspective = value;
+    _hasScrolled();
+  }
+
+  /// The size of the children along the main axis. Children [RenderBox]es will
+  /// be given the [BoxConstraints] to be this exact size.
+  double get itemExtent => _itemExtent;
+  double _itemExtent;
+  set itemExtent(double value) {
+    assert(value != null);
+    assert(value > 0.0);
+    if (value == _itemExtent)
+      return;
+    _itemExtent = value;
+    markNeedsLayout();
+  }
+
+  bool _clipToSize;
```

Comment:
missing getters here and below

Reply:
I was curious what was the rule for having public getters or not. I don't expect anything to ever read from this.

Reply:
The rule is that if you can write:
```dart
  x.y = z;
```
...then you should also be able to write:
```dart
  if (x.y == z) // ...
```
In other words, every public setter should always have a getter. Properties can be read-only or read-write but never write-only.

Anyone who uses the RenderListWheelViewport class directly instead of using the widgets layer is going to quite likely want to use the getters.

Reply:
Done